code/modules/reagents/machinery/reagentgrinder.dm

```diff
@@ -399,43 +406,30 @@ var/global/list/juice_items = list (
 		holdingitems -= O
 	holdingitems = list()
 
-/obj/machinery/reagentgrinder/proc/is_allowed(var/obj/item/weapon/reagent_containers/O)
-	for (var/i in blend_items)
-		if(istype(O, i))
-			return 1
-	return 0
-
-/obj/machinery/reagentgrinder/proc/get_allowed_by_id(var/obj/item/weapon/grown/O)
+/obj/item/proc/get_allowed_by_id()
```

Comment:
I was wrong about absolute/relative typepaths before, those were straight up global procs. Reducing their scope to obj/item works. I'd also rename this proc, since "get_allowed_by_id" doesn't really describe this. Get_allowed_blend_by_id would be great.

Reply:
reworked these out anyways